### ISSUE TITLE
⛔ 🐞 Bug Report: ACF URL Field does not apply actual link to Elementor Gallery images

### ISSUE THREAD

**Issue by ronaldtveen** [elementor/elementor#20738]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

I'm setting up a clients-logo gallery in Elementor and I would like to link each logo image to the clients' website address. This does not seem to work, the link is never applied as a href attribute to the anchor. Strange thing is that when I add an URL in the "fallback" input field, that actually works.

### Steps to reproduce

1. I've setup a new ACF group just if image attachments as described in the [ACF knowledge base](https://www.advancedcustomfields.com/resources/adding-fields-media-attachments/#adding-fields).

1. On a set of logo images in my media library I've added URL's to the images by filling in the ACF field (the full web address, including `https://`).\
[Screenshot](https://res.cloudinary.com/comsi-rtv/image/upload/v1670937983/ACF/acf-logo-field.png)

1. On the Elementor "Our clients"-page I embed the base element "Gallery" with a set of images that include a link to their external website.\
[Screenshot](https://res.cloudinary.com/comsi-rtv/image/upload/v1670938314/ACF/acf-logo-field-setting.png)

1. On the frontend it never actually shows the href attribute on the anchor.
[Screenshot](https://res.cloudinary.com/comsi-rtv/image/upload/v1671004002/ACF/acf-logo-href-missing.png)

1. When I change the settings of point 3. above though and add a fallback-url, that _does_ actually show the fallback as an anchor href-attribute on every single image (so also the images that have an actual custom URL set and should _not_ fall back to the fallback).

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

> Disclaimer, domainnames are redacted...
					
== Server Environment ==

	Operating System: Linux
	Software: Apache/2
	MySQL version: MariaDB Server v10.5.18
	PHP Version: 8.1.13
	PHP Memory Limit: 128M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 26M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==

	Version: 6.1.1
	Site URL: https://testing.comsi.test
	Home URL: https://testing.comsi.test
	WP Multisite: No
	Max Upload Size: 26 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: Europe/Amsterdam
	Admin Email: redacted@comsi.nl
	Debug Mode: Inactive

== Theme ==

	Name: the7dtchild
	Version: 1.0.0
	Author: Dream-Theme
	Child Theme: Yes
	Parent Theme Name: The7
	Parent Theme Version: 11.2.2
	Parent Theme Author: Dream-Theme

== User ==

	Role: administrator
	WP Profile lang: nl_NL
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36

== Active Plugins ==

	ACF Conditional Taxonomy Rules
		Version: 3.0.1
		Author: mattkeys

	Advanced Custom Fields
		Version: 6.0.5
		Author: WP Engine

	Cache Enabler
		Version: 1.8.12
		Author: KeyCDN

	Classic Editor
		Version: 1.6.2
		Author: WordPress Contributors

	ComSi Custom Login
		Version: 1.5.1
		Author: ComSi

	ComSi Translation
		Version: 1.3.5
		Author: ComSi

	DynamicConditions
		Version: 1.6.0
		Author: RTO GmbH

	DynamicTags
		Version: 1.2.1
		Author: RTO GmbH

	Elementor
		Version: 3.9.0
		Author: Elementor.com

	Filter Everything — WooCoomerce Product & WordPress Filter
		Version: 1.7.1
		Author: Andrii Stepasiuk

	MainWP Child
		Version: 4.3.0.1
		Author: MainWP

	PRO Elements
		Version: 3.8.0
		Author: PROElements.org

	Rank Math SEO
		Version: 1.0.103.1
		Author: Rank Math

	The7 Elements
		Version: 2.7.1
		Author: Dream-Theme

	WP Media folder
		Version: 5.5.1
		Author: Joomunited

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast

== Elementor Experiments ==

	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Inactive by default
	Import Export Website Kit: Active by default
	Hide native WordPress widgets from search results: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Default to New Theme Builder: Active by default
	Landing Pages: Active by default
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Lazy Load Background Images: Inactive by default
	Save as Default: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Inactive by default
	Form Submissions: Active by default
	Scroll Snap: Active by default

== Log ==
	
```JS: showing 14 of 14JS: 2022-09-23 08:39:36 [error X 37][https://testing.comsi.test/wp-admin/post.php?post=417&action=elementor#:1:1] Identifier \'$dcWidgetIdInput\' has already been declared 
JS: 2022-09-23 14:36:12 [error X 7][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view307\") has already been destroyed and cannot be used. 
JS: 2022-09-26 07:44:05 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view305\") has already been destroyed and cannot be used. 
JS: 2022-09-26 07:45:10 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view306\") has already been destroyed and cannot be used. 
JS: 2022-09-26 07:45:26 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view2599\") has already been destroyed and cannot be used. 
JS: 2022-09-26 08:10:34 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view299\") has already been destroyed and cannot be used. 
JS: 2022-09-26 09:32:26 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view295\") has already been destroyed and cannot be used. 
JS: 2022-09-26 09:34:52 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view298\") has already been destroyed and cannot be used. 
JS: 2022-09-26 11:52:34 [error X 2][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view313\") has already been destroyed and cannot be used. 
JS: 2022-09-26 11:58:37 [error X 1][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view304\") has already been destroyed and cannot be used. 
JS: 2022-09-28 10:22:16 [error X 5][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view334\") has already been destroyed and cannot be used. 
JS: 2022-11-11 13:04:01 [error X 8][https://testing.comsi.test/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.8.0:3:1061530] Cannot read properties of null (reading \'getBoundingClientRect\') 
JS: 2022-12-13 09:54:44 [error X 2][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view340\") has already been destroyed and cannot be used. 
JS: 2022-12-13 10:47:59 [error X 3][https://testing.comsi.test/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view287\") has already been destroyed and cannot be used. 

Log: showing 20 of 272022-10-31 11:06:14 [info] Elementor/Upgrades - _on_each_version Finished 
2022-10-31 11:06:14 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Start  
2022-10-31 11:06:14 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Finished 
2022-10-31 11:06:14 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.8.0',
)]
2022-10-31 11:06:14 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.8.0',
)]
2022-11-04 14:56:54 [info] elementor-pro::elementor_pro_updater Started 
2022-11-04 14:56:54 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-11-04 14:56:54 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-11-04 14:56:54 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.3',
  'to' => '3.8.0',
)]
2022-11-04 14:56:54 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.3',
  'to' => '3.8.0',
)]
2022-11-14 14:35:34 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-11-14 14:35:37 [info] elementor::elementor_updater Started 
2022-11-14 14:35:37 [info] Elementor/Upgrades - _on_each_version Start  
2022-11-14 14:35:37 [info] Elementor/Upgrades - _on_each_version Finished 
2022-11-14 14:35:37 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.0',
  'to' => '3.8.1',
)]
2022-12-07 09:22:49 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.1',
  'to' => '3.9.0',
)]
2022-12-07 09:22:49 [info] elementor::elementor_updater Started 
2022-12-07 09:22:50 [info] Elementor/Upgrades - _on_each_version Start  
2022-12-07 09:22:50 [info] Elementor/Upgrades - _on_each_version Finished 
2022-12-07 09:22:50 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.8.1',
  'to' => '3.9.0',
)]

PHP: showing 1 of 1PHP: 2022-10-21 16:09:40 [warning X 7][/home/comsi/domains/testing.comsi.test/public_html/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	PRO Elements: Incompatible

== Elementor Pro - Compatibility Tag ==

**Comment by nicholaszein** [comment]
Hello @ronaldtveen,

Thank you for reaching out! 🙏 

# ⛔ While reviewing your report we concluded this is not an issue caused by Elementor. Here's what we found out:

This will be kind of difficult to understand, but I promise I'll try my best to make it as simple as possible.

TL;DR: The method you're using will not work in the way you're applying it because the current queried post/page/CPT (let's call it a _document_) is not the media library item.

## 1. Using ACF:
When you use the Dynamic-tag option in any _document_, you're essentially fetching additional dynamic information directly from the current _document_, which does not include any information about other types of content inside the current document, including media library items. When you use ACF, you have the option to choose what type of content the Field Group will be applied to. In this case, you choose to be applied to media library items (which are also a type of _document_), not to your _document_ type.

## 2. Using Elementor's Dynamic Tags:
Elementor Dynamic Tags will **<ins>only</ins>** check for the metadata related to the current _document_. If you add media library items to that document, it can only see what is built in the default fields of the media library items, not anything else you may add to them, like ACF fields. 

If you select your specific field, it will look for it inside your _document_, not inside the media library items. You can only fetch metadata from the currently queried _document_, unless you set otherwise. Elementor is not setting any other query for that. It only fetches the essential information added to the media library items.

This is the expected default behavior throughout WordPress.

# ✅ What you can do to solve your issue:
There are a number of ways to accomplish what you want, here are a few of them:
1. Create a CPT called "Clients", include your ACF link field, add their logos as featured images, add the URL in the ACF field, and use Elementor Pro's new Loop Grid widget to query the "Clients" CPT. Then add an Image widget fetching the featured image of your CPT, and set the link as your ACF field.
2. Use a 3rd party add-on like Dynamic Content to create your gallery.
3. Do it manually.
4. Use a dedicated gallery plugin with support for ACF fields.

For now, that's all we can help with. If you think this is still an issue caused by Elementor, please feel free to open a [new issue report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml) and provide us with the most information you can possibly get by following our guidelines:

* Please create GitHub issues **only for Elementor bugs**. GitHub issues ARE NOT FOR SUPPORT or for issues with 3rd party plugins.
* To **avoid duplicates**, please search this repository for similar issues before submitting your own.
* If the issue can't be recreated on other sites, it's not an Elementor bug. In that case, **use the support links below before submitting your issue here**.
* Make sure you are using the most **updated versions** of Elementor, WordPress & your theme.
* Deactivate all of your plugins. If this solves the problem, gradually activate your plugins one by one, until you spot the problematic plugin, and then **contact the 3rd party plugin developer**.
* Make sure to **provide a detailed description** of your issue.
* Make sure to **provide detailed steps to replicate** your issue.
* Make sure the issue occurs with **only Elementor** (and Elementor Pro) plugin(s) active.
* Make sure the issue happens with a **blank theme** active ([Hello Elementor](https://wordpress.org/themes/hello-elementor/)).
* Make sure to **provide a copy of your _Elementor > System Info_**. Download the `.txt` file and drag it inside the "System Info" field when submitting a new bug report.

### ⚠️ Please Note that GitHub issues are reserved for 🐞**Bug Reports**, and should not be used for support.

❔ If you have questions or need general support, please use one of the following channels:

- [Elementor Global Community](https://facebook.com/groups/Elementors/) on Facebook
- [Elementor Help Center](https://docs.elementor.com)
- [Support Forum](https://wordpress.org/support/plugin/elementor) on WordPress.org 
- [Elementor Developers Documentation](https://developers.elementor.com/)

💎 If you have purchased [Elementor Pro](https://elementor.com/pro), you are entitled to personal support, please see your purchase email or visit your [Elementor Account page](https://my.elementor.com/) for details.

I hope this helps.

Kind regards
